File: app/src/main/java/br/com/arcom/signpad/data/AppDataBase.java
package br.com.arcom.signpad.data;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import br.com.arcom.signpad.utilities.Constantes;

import static br.com.arcom.signpad.utilities.Constantes.DATABASE_VERSION;

@Database(
        entities = {
                SigaToken.class,
                LgpdVisitante.class
        }
        , version = DATABASE_VERSION)
@TypeConverters({Converters.class})
public abstract class AppDataBase extends RoomDatabase {

    private static AppDataBase dataBase;

    public synchronized static AppDataBase getInstance(Context context) {
        if (dataBase == null || !dataBase.isOpen()) {
            dataBase = Room.databaseBuilder(context.getApplicationContext()
                    , AppDataBase.class, Constantes.DATABASE_NAME)
                    .allowMainThreadQueries()
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return dataBase;
    }

    public abstract SigaTokenDAO sigaTokenDAO();

    public abstract LgpdVisitanteDAO lgpdVisitanteDAO();

}

File: app/src/main/java/br/com/reconhecimento/database/db/AppDatabase.kt
package br.com.reconhecimento.database.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import br.com.reconhecimento.data.model.FuncaoAtivacao
import br.com.reconhecimento.database.converter.AppTypeConverters
import br.com.reconhecimento.database.dao.ErroDao
import br.com.reconhecimento.database.dao.FuncaoAtivacaoDao
import br.com.reconhecimento.database.dao.TreinosDao
import br.com.reconhecimento.database.entity.ErroEntity
import br.com.reconhecimento.database.entity.FuncaoAtivacaoEntity
import br.com.reconhecimento.database.entity.TreinosEntity
import br.com.reconhecimento.util.DATABASE_NAME

@Database(
    entities = [TreinosEntity::class, ErroEntity::class, FuncaoAtivacaoEntity::class],
    version = 1, exportSchema = false
)
@TypeConverters(AppTypeConverters::class)
abstract class AppDatabase : RoomDatabase() {

    abstract fun treinosDao(): TreinosDao
    abstract fun erroDao(): ErroDao
    abstract fun funcaoAtivacaoDao(): FuncaoAtivacaoDao

    companion object {

        @Volatile
        private var instance: AppDatabase? = null

        fun getInstance(context: Context): AppDatabase {
            return instance ?: synchronized(this) {
                instance ?: buildDatabase(context).also { instance = it }
            }
        }

        private fun buildDatabase(context: Context): AppDatabase {
            return Room.databaseBuilder(context, AppDatabase::class.java, DATABASE_NAME)
                .fallbackToDestructiveMigration()
                .build()
        }
    }
}